### PR TITLE
feat: Added pdf and cdf of some probability distributions

### PR DESCRIPTION
**Pull request by ttuanho**
#### Description of Change

- Added the cdf and pdf function of the following distributions:
   - Poisson distribution
   - Normal distribution
   - Uniform distribution
   - Exponential distribution

#### Checklist

- [x] Added description of change
    - Some were tested against other calculator
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example
     - need more examples for other distributions
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1143"></a>

### CODE REVIEW

**Comment by ttuanho** [comment]
Appreciate if someone can help me with the [building error on Windows CI](https://github.com/TheAlgorithms/C-Plus-Plus/runs/1197800749?check_suite_focus=true). 

Any tips for someone with different environment (MacOS)? Or should I push to the repo CI for every time I made a change?

File: probability/distributions.cpp [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+ *  </li>
+ * </ul>
+ */
+
+#include <cassert>
+#include <cmath>
+#include <iostream>
+#include <vector>
+
+#define _USE_MATH_DEFINES
```
Comment (kvedala) [inline probability/distributions.cpp:26]: this should be defined **before** including the `cmath` header.

File: probability/distributions.cpp [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+constexpr double E = 2.71828182845904523536;
```
Comment (kvedala) [inline probability/distributions.cpp:28]: this is also available in `cmath` header as `M_E`.

C++20 has a dedicated STL class `std::number` containing precision specific implementations. But we are using C++11. 

Other standard constants: https://www.gnu.org/software/libc/manual/html_node/Mathematical-Constants.html

File: probability/distributions.cpp [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+using ll = int64_t;
+
+// Find the greatest common divisor
+ll gcd(ll a, ll b) { return b ? gcd(b, a % b) : a; }
```
Comment (kvedala) [inline probability/distributions.cpp:33]: please make this function static inside the `Util` namespace.

File: probability/distributions.cpp [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+ * \end{cases}
+ * \f$
+*/
+double pdf(double x, double lambda) {
+    double ans = (x >= 0) ? lambda * std::exp(-lambda * x) : 0;    
+    return ans;
+}
+}  // namespace Exp
+
+namespace Gauss {}  // namespace Gauss
```
Comment (kvedala) [inline probability/distributions.cpp:264]: can this empty namespace be removed?

File: probability/distributions.cpp [hunk 5]
```diff
@@ -0,0 +1,311 @@
+/**
+ * \file
+ * \brief Implementation of Discrete & Continuous Distributions
+ * \author [Tuan Ho](https://github.com/ttuanho)
+ * \note The implementation is based on [this
+ * cheatsheet](https://web.cs.elte.hu/~mesti/valszam/kepletek)
+ * \todo
+ * <ul>
```
Comment (kvedala) [inline probability/distributions.cpp:8]: This is good documentation. you can also use the simple syntax of Markdown to format the documentation just like you used for the links.

**Review by kvedala — CHANGES_REQUESTED** [review]
:+1: Good work. Thank you for your contribution

File: probability/distributions.cpp (hunk 2 above)
Comment (ttuanho) [reply probability/distributions.cpp]: turned out I didn't use this const but thanks for the heads up

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/ttuanho/C-Plus-Plus/actions

**Review by kvedala — CHANGES_REQUESTED** [review]
Please fix this error and the warnings related to the file. 
image

**Pull request closed without merging**